downloads/swan-lake-release-notes/swan-lake-preview8/RELEASE_NOTE.md

```diff
@@ -0,0 +1,519 @@
+
+```ballerina
+int a\-b;
+int 'a\-b;
+```
+
+##### Included Record Parameters
+
+Included record parameters can be specified as `*T P` in which `T` denotes the record type descriptor and `P` denotes the name of the parameter. When this is compared with the required parameter, the difference is that the values, which are used for the fields of the included record parameter behave as named arguments for the function call, if it had been declared as parameters of the function.
+
+The field names of all included record parameters, which are not of the `never` type, must be distinct from each other and also from the names of the parameters.
+
+```ballerina
+type Student record {
+	string firstName;
+	string secondName;
+	int age;
+	Address address?;
+	never gender?;
+};
+
+type Address record {|
+	string city;
+	string country;
+|};
+
+type Grades record {|
+	never maths?;
+	never physics?;
+	never chemistry?;
+
+	int...;
+|};
+
+function studentDetails(int addmissionNo, string addmissionDate, *Student student) {
+	string fullName = <string>student.firstName  + <string>student.secondName;
+	int age = student.age;
+	if (student["address"] is Address) {
+    	Address address = <Address>student["address"];
+	}
+}   
+
+function studentTotalMarks(int maths, int physics, int chemistry, *Grades grades) {
+	int totalMarks = maths + physics + chemistry;
+	foreach [string, int?] [subject, marks] in grades.entries() {
+    	totalMarks = totalMarks + <int>marks;
+	}
+}
+```
+
+A named argument in a function call can correspond to an included record parameter in two different ways.
+
+1. Being relevant to a field of the included recorded parameter, which is not of the `never` type.
+
+    ```ballerina 
+    public function main() {
+        studentDetails(1001, "2020-Nov-20", firstName = "Peter", secondName = "So", age = 18);
+        studentDetails(1002, "2020-Nov-22", firstName = "Anne", secondName = "Doe", age = 20, address = {city: "Colombo", country: "Sri Lanka"});
+    }
+    ```
+
+2. Being relevant to a record rest descriptor of an included record parameter. Here, the field is described by a record rest descriptor either as explicitly part of an exclusive record type descriptor or as implicitly as part of an inclusive record type descriptor. The two conditions below should be satisfied for this.
+
+    - In the parameter list, there should be only one included record parameter with a record rest descriptor.
+    - The included record parameter with the record rest descriptor must include optional individual field descriptors of the type `never`, which correspond to each parameter name and the names of each individual field descriptor of the other included record parameters. In addition to these optional individual field descriptors, there should not be any other field descriptors with this included record parameter.
```

Comment:
```suggestion
    - The open record type must disallow fields of the same names as the other parameters and individual field descriptors of the other included record parameters, by including optional individual field descriptors of the type `never`. 
```

@chiranSachintha, do we need "In addition to these optional individual field descriptors, there should not be any other field descriptors with this included record parameter". This kinda implies the following cannot be used as an included-record-param.

```ballerina
type Grades record {|
    never math?;
    int physics;
    int...;
|};
```

Reply:
In this case, named argument belongs to an included record parameter as an additional field(allowed by the record being open). So there can be only fields with never type within the included record parameter.

Reply:
Ah, missed that. 

How about
```suggestion
    - The open record type must disallow fields of the same names as the other parameters and individual field descriptors of the other included record parameters, by including optional individual field descriptors of the type `never`.  In addition to these optional individual field descriptors, there should not be any other field descriptors in this record type.
```

Reply:
Included in https://github.com/praneesha/ballerina-dev-website/pull/10